WMF Framework/Router.swift

```diff
@@ -0,0 +1,165 @@
+@objc(WMFRouter)
+public class Router: NSObject {
+    public enum Destination: Equatable {
+        case inAppLink(_: URL)
+        case externalLink(_: URL)
+        case article(_: URL)
+        case articleHistory(_: URL, articleTitle: String)
+        case articleDiffCompare(_: URL, fromRevID: Int?, toRevID: Int?)
+        case articleDiffSingle(_: URL, fromRevID: Int?, toRevID: Int?)
+        case userTalk(_: URL)
+        case search(_: URL, term: String?)
+    }
+    
+    unowned let configuration: Configuration
+    required init(configuration: Configuration) {
+        self.configuration = configuration
+    }
+    
+    // From https://github.com/wikimedia/mediawiki-title
+    private let namespaceRegex = try! NSRegularExpression(pattern: "^(.+?)_*:_*(.*)$")
+    private let mobilediffRegexCompare = try! NSRegularExpression(pattern: "^mobilediff/([0-9]+)\\.\\.\\.([0-9]+)", options: .caseInsensitive)
+    private let mobilediffRegexSingle = try! NSRegularExpression(pattern: "^mobilediff/([0-9]+)", options: .caseInsensitive)
+    private let historyRegex = try! NSRegularExpression(pattern: "^history/(.*)", options: .caseInsensitive)
+    
+     internal func destinationForWikiResourceURL(_ url: URL) -> Destination? {
+        
+        if configuration.isWResource(url) {
```

Comment:
Figured out why I had a hard time testing - I think we will need to update our [association](https://en.wikipedia.org/apple-app-site-association) file in order for these /w/ path [links](https://phabricator.wikimedia.org/T228789) to intercept. 😕

Reply:
Good catch! We can edit that file in the [operations/mediawiki-config repo](https://gerrit.wikimedia.org/r/plugins/gitiles/operations/mediawiki-config/+/refs/heads/master/docroot/wikipedia.org/apple-app-site-association). My guess is we'll release this with support then update the file sometime after enough users have upgraded. I'll confirm there's no other reason those URLs were excluded